javascript/app.js

```diff
@@ -137,8 +144,23 @@ FinancialFreedom.controller('TimeToRetirementController', ['$scope', 'Retirement
     
 }]);
 
-FinancialFreedom.filter('percentage', ['$filter', function ($filter) {
-  return function (input, decimals) {
-    return $filter('number')(input * 100, decimals) + '%';
-  };
-}]);
+FinancialFreedom.directive("percent", function($filter){
+    var p = function(viewValue){ // format model value
+        var m = viewValue.match(/^(\d+)\/(\d+)/);
+        if (m != null)
+          return $filter('number')(parseInt(m[1])/parseInt(m[2]), 1);
+        return $filter('number')(parseFloat(viewValue)/100, 1);
+    };
+
+    var f = function(modelValue){ // format display value
+        return $filter('number')(parseFloat(modelValue)*100, 2) + '%';
+    };
+    
+    return {
+        require: 'ngModel',
+        link: function(scope, ele, attr, ctrl){
+            ctrl.$parsers.unshift(p);
+            ctrl.$formatters.unshift(f);

```

Comment:
Ok, that's ok I guess. Typically it's good to at least have a general idea of what code is doing before copy pasting it into one's project, but you're just starting out and this isn't super vital so it's not a big deal.